pkg/apis/walruscore/v1/resource_run.go

```diff
@@ -41,3 +109,191 @@ type ResourceRunList struct {
 }
 
 var _ runtime.Object = (*ResourceRunList)(nil)
+
+// ResourceRunStepType describes the type of the resource run step.
+// +enum
+type ResourceRunStepType string
+
+const (
+	// ResourceRunStepTypePlan is the type of the plan resource, it is system generated.
+	ResourceRunStepTypePlan ResourceRunStepType = "Plan"
+	// ResourceRunStepTypeApply is the type of the apply resource, it is system generated.
+	ResourceRunStepTypeApply ResourceRunStepType = "Apply"
+
+	// ResourceRunStepTypeContainer is the type of the container resource, it depends on the resource run hooks and resource run template.
+	ResourceRunStepTypeContainer ResourceRunStepType = "Container"
+	// ResourceRunStepTypeApproval is the type of the approval resource, it depends on the resource run hooks and resource run template.
+	ResourceRunStepTypeApproval ResourceRunStepType = "Approval"
+	ResourceRunStepTypeUnknown  ResourceRunStepType = "Unknown"
+)
+
+// ResourceRunStep stores the step run result of the resource run.
+// The step may be the plan, apply, and hook step defined in the resource hook or resource run template.
+type ResourceRunStep struct {
+	// StatusDescriptor defines the status of the step.
+	StatusDescriptor `json:",inline"`
+
+	// Name is the name of the step.
+	Name string `json:"name"`
+
+	// Type is the type of the step.
+	Type ResourceRunStepType `json:"type"`
+
+	// ResourceRunStepTemplate is a reference to a resource step template.
+	ResourceRunStepTemplate *ResourceRunStepTemplateRef `json:"runStepTemplate,omitempty"`
+
+	// StartedAt is the time when the step started.
+	StartedAt meta.Time `json:"startedAt"`
+
+	// FinishedAt is the time when the step finished.
+	FinishedAt meta.Time `json:"finishedAt"`
+
+	// Duration is the duration of the step.
+	Duration *int64 `json:"duration"`
```

Comment:
what's the unit of the Duration? if in seconds, please use DurationInSeconds, otherwise, make its type meta.Duration.